crates/net/rpc-builder/src/lib.rs

```diff
@@ -92,9 +92,12 @@ where
     /// Configures the [RpcModule] which can be used to start the server(s).
     ///
     /// See also [RpcServer::start]
-    pub fn build(self) -> RpcModule<()> {
+    pub fn build(self, _servers: Vec<RpcServer>) -> Vec<RpcModule<()>> {
```

Comment:
this goes in the right direction but is yet exactly what we need.

what we want here, instead of `Vec<RpcServer>` is a type that has `Option<RpcModuleConfig>` per transport. 
something like

```rust
struct TransportRpcModuleConfig {
  ws: Option<RpcModuleConfig>,
  HTTP, ipc
}
```

then build should return a similar type
```rust
struct TransportRpcModules {
  ws: Option<RpcModule>
 ....
} 
```

Reply:
For this, we'll have to parse something like ```--http.api eth,debug --ws-.api eth``` into a ```(http, [eth, debug]), (ws, [eth])``` type struct, that constructs ```TransportRpcModuleConfig```.